### PR TITLE
Add ignore: ["first-nested"] options

### LINKED ISSUE #3178: Add ignore: ["first-nested"] to *-empty-line-before
The `"first-nested"` option was added to `custom-property-empty-line-before` and `declaration-empty-line-before` in 9.0.0. Whilst configuring my config for compatibility with prettier, I realised the same option is needed for `at-rule-empty-line-before` and `rule-empty-line-before` as prettier also formats the empty line before the first nested instances of these.

### PR DESCRIPTION
**Pull request by jeddy3**
> Which issue, if any, is this issue related to?

Closes #3178 

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

**Comment by jeddy3** [comment]
>  -   Added: `ignore: ["first-nested"]` to `at-rule-empty-line-before` ([#3179](https://github.com/stylelint/stylelint/pull/3179)).
>  -   Added: `ignore: ["first-nested"]` to `rule-empty-line-before` ([#3179](https://github.com/stylelint/stylelint/pull/3179)).

File: lib/rules/rule-empty-line-before/README.md [hunk 1]
```diff
@@ -262,6 +262,22 @@ The following patterns are *not* considered violations:
 a {}
 ```
 
+#### `"after-comment"`
```
Comment (AlexWayfer) [inline lib/rules/rule-empty-line-before/README.md]: Wrong!

Comment (jeddy3) [reply lib/rules/rule-empty-line-before/README.md]: Good spot. Would you like to create to PR to fix this?

**Comment by AlexWayfer** [comment]
What's the difference with existing [`except: ["first-nested"]` in `at-rule-empty-line-before`](https://stylelint.io/user-guide/rules/at-rule-empty-line-before/#first-nested) and [the same in `rule-empty-line-before`](https://stylelint.io/user-guide/rules/rule-empty-line-before/#first-nested)?

File: lib/rules/rule-empty-line-before/README.md (hunk 1 above)
Comment (AlexWayfer) [reply lib/rules/rule-empty-line-before/README.md]: I think I can. But I'm still [not sure that this PR is necessary](https://github.com/stylelint/stylelint/pull/3179#issuecomment-367324023).

**Comment by jeddy3** [comment]
From [here](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#secondary):

> - Use `"ignore"` when you want the rule to simply skip-over a particular pattern.
> - Use `"except"` when you want to invert the primary option for a particular pattern.

The `ignore` option is used when you don't want stylelint to enforce anything for that pattern.

**Comment by AlexWayfer** [comment]
Thank you, @jeddy3.

Are there cases (examples) for these rules? I see the issue, but I don't see examples, just "let's make it here like in other rules".

**Comment by jeddy3** [comment]
> I see the issue, but I don't see examples, just "let's make it here like in other rules".

An option is generally added to a rule when someone has made a specific request and stated their use case e.g. here's the issue that requests the [`except: ["after-same-name"]`](https://github.com/stylelint/stylelint/issues/2211) option for `at-rule-empty-line-before`.

> Are there cases (examples) for these rules? 

The use case of the new `ignore` options added in this PR can be found in the original issue (https://github.com/stylelint/stylelint/issues/3178) that this PR closes. It's a bit brief I'm afraid, so I understand the confusion. A more detailed use case [can be found in the issue](https://github.com/stylelint/stylelint/issues/3101#issuecomment-355338437) that requests the `ignore: ["first-nested"]` option for `declaration-empty-line-before` i.e. these options are useful if you use another tool, e.g. prettier, to pretty-print the empty-line before the first nested nodes. 

(I've added a link from the original issue to this more detailed use case to make it easier for future users to understand why these two new options were added).

> Small explanation: we have `except: ["first-nested"]` in our projects, and I'm not sure, should we switch to `ignore` or not.

If you want _stylelint to enforce_ a convention for the first nest node, then stick with `except: ["first-nested"]`. If you want to stylelint to ignore the first nest node because, perhaps, you use _another tool to enforce_ a convention, then use `ignore`.

**Comment by AlexWayfer** [comment]
Thank you.

Why `except` option is not named as `invert`? :grimacing: 

I think, "enforce a convention" and "invert the primary option" are pretty different means.

I think I still can't understand `except` option, but I understood `ignore` option from the first time.

But OK, I think it's my bad. I understood that if you have `always` as primary option, then `except` will set `never` for listed cases, and vice-versa. And `ignore` doesn't set any pimary option for listed cases at all.

**Comment by jeddy3** [comment]
> Why `except` option is not named as `invert`? 😬

We have to [go back a couple of years](https://github.com/stylelint/stylelint/issues/207) to answer that one... it looks like we borrowed the word from eslint. 

I think it still works well enough, though. Take the `indentation` rule for example:

```json
"indentation": [2, { "except": ["param"] } ]
```

i.e. indent by 2 spaces except for params.

Unlike words like "always" and "never", there isn't an intuitive inverted value for the number 2. So, "except" might not be the perfect word but it works well enough in a fair few contexts.

> But OK, I think it's my bad. I understood that if you have `always` as primary option, then `except` will set `never` for listed cases, and vice-versa. And `ignore` doesn't set any pimary option for listed cases at all.

Yes, that's correct.

**Pull request merged**